src/shogun/distributions/HMM.h

```diff
@@ -1240,8 +1240,12 @@ class HMM : public RandomMixin<Distribution>
 		float64_t mod_prob;
 
 		/// true if model probability is up to date
+		#ifdef USE_HMMPARALLEL
+		std::atomic<bool> mod_prob_updated;
```

Comment:
@vigsterkr not sure who the expert on this algo is, but I am pretty sure if you are using multiple threads this variable causes a race condition

Reply:
git blame? but honestly i think this algo has not been touched and only lingered around with minor fixups for a loooooong time

Reply:
indeed, last time this seemed to have been actively developed was 2010 or so :D